add: malware (#217)
cloudsmith-io/cloudsmith-docs: src/content/supply-chain-security/malware-detection/malicious-packages.mdx

- **Dependency Confusion**: This technique targets corporate build systems. An attacker discovers the name of a private, internal package (e.g., `acme-auth`) and publishes a malicious package with the same name to a public registry, often with a higher version number, tricking automated tools into downloading it.
- **Account Takeover**: Attackers compromise the account of a legitimate package maintainer. They then publish a new, malicious version of the trusted package. For instance, if the maintainer of `super-utils` gets their account stolen, an attacker could release version `1.2.4` containing code to steal credit card information.
- **Protestware/Malicious Maintainers**: Sometimes, the threat comes from the package maintainers themselves. As an act of protest or malice, they might add harmful code to their own popular library. For example, the maintainer of `node-ipc` added code that would delete files on systems with specific IP addresses as a form of protest.

> - **Registry T

Reviewer: Looks like a bit for registry typo squatting that was mentioned above!

Author: yep thanks! :)

OSV.dev, initiated by Google, provides a unified, open, and distributed database for vulnerability information, making it easy for developers and automated security tools to consume. By cataloging malicious packages within OSV.dev, the OpenSSF project centralizes intelligence that helps developers and organizations quickly identify threats and strengthen their software supply chains.

## Malicious package detection in Cloudsmith

<Note variant="note" headline="Plans">
Malicious package detection via OSV.dev is available for Ultra and Enterprise customers.
</Note>

The malicious packages dataset is integrated into [Enterprise Policy Management](/supply-chain-security/epm) and [Continuous Security](/supply-chain-security/continuous-security) (both in Early Access). This allows you to define policies that automatically take action when a malicious package is detected.

> For example, here we can see an EPM policy that quarantines any package that is considered malicious in the OSV database (and tags it with `MALICIOUS_PACKAGE`):
> 
> <BlockImage src={policy_malicious_packages} alt=""></BlockImage>
> 
> If you upload a package flagged as malicious, the policy automatically triggers and quarantines it.

Reviewer 2: ```suggestion
Here is an EPM policy that matches any package which has been flagged as malicious by Continuous Security, quarantines it, and adds a `MALICIOUS_PACKAGE` tag:

<BlockImage src={policy_malicious_packages} alt=""></BlockImage>
```

<BlockImage src={malicious_package} alt=""></BlockImage>

### Example Rego policy

> To flag any artifact considered malicious in Cloudsmith, check whether the`vulnerability.id` field starts with the prefix `MAL-` - all malicious packages are identified with this string. 

Reviewer 2: ```suggestion
To target malicious packages in policy-as-code, check whether the`vulnerability.id` field starts with the prefix `MAL-` - all malicious packages are identified with this string. 
```

Here’s an example policy:

```rego
package cloudsmith

default match := false

match if count(malicious_packages) > 0

malicious_packages := [vulnerability.id |
	some vulnerability in input.v0.osv
	startswith(vulnerability.id, "MAL-")
]
```

### Supported Formats

Below you can find the package [formats supported](https://github.com/ossf/malicious-packages/tree/main/osv/malicious) by the OSSF Malicious Packages project:

> | Registry / Manager | Language / Ecosystem |

Reviewer: is hex another supported format?

Author: @chrisimcevoy @clairemcdyre

Author: I think you are right, here: https://osv.dev/list?ecosystem=Hex
